### PR TITLE
Use name validations from our API instead of the k8s core

### PR DESCRIPTION
**Pull request by jpeeler**
Written by Paul, slightly modified by me.
(Supercedes https://github.com/kubernetes-incubator/service-catalog/pull/330)

### CODE REVIEW

**Comment by MHBauer** [comment]
@pmorie I'd like to understand the why behind using `apivalidation.NameIsDNSSubdomain`. I dove into the k8s impl, and I can see that a lot of components in there use this for names, but the code has no comment on why. I can see it ends up being rfc1123. That rfc is 97 pages. I am not familiar with it to know which section to look at.  The word 'subdomain' or any versions of it does not exist in that rfc.

I accept 'k8s does it' as a reason here (in fact, `NameIsDNSSubdomain` is what `ValidateReplicationControllerName` maps to, so this is not a functional change), but I'm asking for a direct link to the exact part of the rfc that defines we're talking about. It would be doubly nice if that link was then shoved into the upstream codebase.

Right now the field might as well be called 'isValidFloobleCrank'.

lgtm

**Comment by pmorie** [comment]
> isValidFloobleCrank

🥇

File: pkg/apis/servicecatalog/validation/binding.go [hunk 1]
```diff
@@ -23,13 +23,17 @@ import (
 	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
 )
 
+// validateBindingName is the validation function for Binding names.
+var validateBindingName = apivalidation.NameIsDNSSubdomain
```
Comment (arschles) [inline pkg/apis/servicecatalog/validation/binding.go]: why put this into a package-level var? it it used anywhere else besides the below?

File: pkg/apis/servicecatalog/validation/broker.go [hunk 2]
```diff
@@ -29,15 +29,19 @@ import (
...
-// ValidateBroker makes sure a broker object is okay?
+// ValidateBroker implements the validation rules for a BrokerResource.
 func ValidateBroker(broker *sc.Broker) field.ErrorList {
 	allErrs := field.ErrorList{}
-	// validate the name?
-	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&broker.ObjectMeta, false, /*namespace*/
-		apivalidation.ValidateReplicationControllerName, // our custom name validator?
-		field.NewPath("metadata"))...)
+
+	allErrs = append(allErrs,
```
Comment (arschles) [inline pkg/apis/servicecatalog/validation/broker.go]: in the event that `apivalidation.ValidateObjectMeta` returns nil, `len(allErrs)` will still be 1. is that acceptable?

File: pkg/apis/servicecatalog/validation/binding.go (hunk 1 above)
Comment (arschles) [inline pkg/apis/servicecatalog/validation/binding.go]: same with the below, similar, package level vars

Comment (pmorie) [inline pkg/apis/servicecatalog/validation/binding.go]: This is all me, following the pattern from upstream.  I can change them to just be the `NameIsDNSSubdomain` - no reason I'm aware of for them to be vars.

File: pkg/apis/servicecatalog/validation/broker.go (hunk 2 above)
Comment (pmorie) [inline pkg/apis/servicecatalog/validation/broker.go]: Disagree: https://play.golang.org/p/EuvD0bQ2pd

Comment (arschles) [inline pkg/apis/servicecatalog/validation/broker.go]: simpler: https://play.golang.org/p/QJ8bN1xayI

Comment (arschles) [inline pkg/apis/servicecatalog/validation/broker.go]: ah, the key is the varargs. I stand corrected. carry on!

**Pull request merged**